udata/core/spam/api.py

```diff
@@ -0,0 +1,58 @@
+from mongoengine import Q
+
+from udata.api import api, API
+from udata.auth import admin_permission
+from udata.core.discussions.models import Discussion
+from udata.core.spam.fields import potential_spam_fields
+from udata.core.spam.models import POTENTIAL_SPAM
+from udata.utils import id_or_404
+
+
+class SpamAPIMixin(API):
+    """
+    Base Spam Model API.
+    """
+    model = None
+
+    def get_model(self, id):
+        """
+        This function returns the base model and the spamable model which can be different. The base model is the
+        model stored inside Mongo and the spamable model is the embed document (for example a comment inside a
+        discussion)
+        """
+        model = self.model.objects.get_or_404(id=id_or_404(id))
+        return model, model
```

Comment:
I'm confused, we're returning model in both cases here

Reply:
The comment above is not enough to explain?